cpp/example_code/transfer-manager/transferOnStream.cpp

```diff
@@ -86,35 +88,53 @@ int main(int argc, char** argv)
 
         auto uploadHandle = transfer_manager->UploadFile(LOCAL_FILE, BUCKET, KEY, "text/plain", Aws::Map<Aws::String, Aws::String>());
         uploadHandle->WaitUntilFinished();
-        std::cout << "File upload finished!" << std::endl;
-
-        // verify upload expected length of data
-        assert(uploadHandle->GetBytesTotalSize() == uploadHandle->GetBytesTransferred());
-        g_file_size = uploadHandle->GetBytesTotalSize();
-
-        // This buffer is what we used to initialize streambuf and is in memory
-        Aws::Utils::Array<unsigned char> buffer(BUFFER_SIZE);
-        auto downloadHandle = transfer_manager->DownloadFile(BUCKET,
-                KEY, 
+        bool success = uploadHandle->GetStatus() == Transfer::TransferStatus::COMPLETED; 
+      
+        if (!success)
+        {
+            auto err = uploadHandle->GetLastError();           
+            std::cout << "File upload failed:  "<< err.GetMessage() << std::endl;
+        }
+        else
+        {
+            std::cout << "File upload finished." << std::endl;
+
+            // Verify upload expected length of data
+            assert(uploadHandle->GetBytesTotalSize() == uploadHandle->GetBytesTransferred());
+            g_file_size = uploadHandle->GetBytesTotalSize();
+
+            // This buffer is what we used to initialize streambuf and is in memory
+            Aws::Utils::Array<unsigned char> buffer(BUFFER_SIZE);
+            auto downloadHandle = transfer_manager->DownloadFile(BUCKET,
+                KEY,
                 [&]() { //create stream lambda fn
                     return Aws::New<MyUnderlyingStream>("TestTag", Aws::New<Stream::PreallocatedStreamBuf>("TestTag", buffer.GetUnderlyingData(), BUFFER_SIZE));
                 });
-        downloadHandle->WaitUntilFinished();
-        std::cout << "File download to memory finished!" << std::endl;
-
-        // verify download expected length of data
-        assert(downloadHandle->GetBytesTotalSize() == downloadHandle->GetBytesTransferred());
-
-        // verify length of upload equals download 
-        assert(uploadHandle->GetBytesTotalSize() == downloadHandle->GetBytesTotalSize());
-
-        // write buffered data to local file copy
-        Aws::OFStream storeFile(LOCAL_FILE_COPY.c_str(), Aws::OFStream::out | Aws::OFStream::trunc);
-        storeFile.write((const char*)(buffer.GetUnderlyingData()), downloadHandle->GetBytesTransferred());
-        storeFile.close();
-        std::cout << "File dumpped to local file finished! You can verify the two files' content using md5sum." << std::endl;
-
-        // verify upload file is the same as downloaded copy. I did that simply using `md5sum file file_copy`
+            downloadHandle->WaitUntilFinished();// block calling thread until download complete
```

Comment:
Block ... complete.

We use full sentences, with punctuation, in comments. I won't repeat this comment from here on.